docs/source/pytest.rst

```diff
@@ -0,0 +1,358 @@
+(it must be located in the root of the project):
+
+.. code:: python
+
+   pytest_plugins = ("cocotb_tools.pytest.plugin",)
+
+By defining the ``PYTEST_PLUGINS`` environment variable:
+
+.. code:: shell
+
+   export PYTEST_PLUGINS="cocotb_tools.pytest.plugin"
+
+By using ``-p <plugin>`` option when invoking `pytest`_ command line interface:
+
+.. code:: shell
+
+   pytest -p cocotb_tools.pytest.plugin ...
+
+Building and Testing
+====================
+
+:py:class:`cocotb_tools.pytest.hdl.HDL` wraps :py:class:`cocotb_tools.runner.Runner`
+allowing to fully configure cocotb runner by using command line arguments ``--cocotb-*``,
+configuration files like `pyproject.toml`_ or `fixture`_ arguments.
+
+Plugin includes own ``hdl`` fixture that will create new instance of :py:class:`cocotb_tools.pytest.hdl.HDL`
+with `pytest`_ built-in `request`_ fixture that is providing information of the requesting test function
+including current configuration of `pytest`_.
+
+Example content of ``conftest.py`` file:
+
+.. code:: python
+
+   import pytest
+   from cocotb_tools.pytest.hdl import HDL
+
+
+   @pytest.fixture(name="sample_module")
+   def sample_module_fixture(hdl: HDL) -> HDL:
+       """Define HDL design by adding HDL source files to it.
+
+       Args:
+           hdl: Fixture created by cocotb plugin, representing HDL design.
+
+       Returns:
+           Representation of HDL design with added HDL source files.
+       """
+       hdl.sources = (
+           # List HDL source files,
+           "sample_module.sv",
+       )
+
+       return hdl
+
+
+Example content of ``test_sample_module.py`` file:
+
+.. code:: python
+
+   import pytest
+   from cocotb_tools.pytest.hdl import HDL
+
+
+   # Without providing positional arguments or test_module option to cocotb decorator,
+   # plugin will use current file as cocotb testbench (Python file with cocotb tests).
+   # If toplevel option was not provided, it will be based on name of first test_module
+   # but with removed test_* prefix or *_test suffix.
+   @pytest.mark.cocotb  # equivalent to @pytest.mark.cocotb("test_dut", toplevel="dut")
+   def test_sample_module(sample_module: HDL) -> None:
+       """Build HDL design and run HDL simulator to execute cocotb tests.
+
+       Args:
+           sample_module: Instance of defined HDL design.
+       """
+       sample_module.test()
+
+
+   # @pytest.mark.cocotb or @cocotb.test decorator is not required if test function
+   # starts with test_* prefix, is coroutine function (``async``) and with ``dut`` argument.
+   async def test_some_dut_feature(dut) -> None:
+        """Cocotb test for DUT."""
+
+@pytest.mark.cocotb
+===================
+
+Provided ``@pytest.mark.cocotb`` marker by :py:mod:`cocotb_tools.pytest.plugin` allows
+to configure all aspects of cocotb test and cocotb runner. Marker recognizes all
+named arguments from :py:func:`cocotb.test` and :py:class:`cocotb_tools.runner.Runner`.
```

Comment:
Line breaks usually occur on sentence or phrase boundaries.
```suggestion
Provided ``@pytest.mark.cocotb`` marker by :py:mod:`cocotb_tools.pytest.plugin` allows to configure all aspects of cocotb test and cocotb runner.
Marker recognizes all named arguments from :py:func:`cocotb.test` and :py:class:`cocotb_tools.runner.Runner`.
```

Reply:
Finally I am *not* a fan of this thing supporting the features of `@cocotb.test`. Most of it's features are better implemented using pytest. Let's remove those.